### PR TITLE
openvino: add 2023.1.0 version

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
Specify library name and version:  **openvino/2023.1.0**

Closes https://github.com/conan-io/conan-center-index/issues/7277

---

- [X] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [X] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/openvino/all/conanfile.py
Comment (prince-chrismc) [inline recipes/openvino/all/conanfile.py:145]: This syntax is valid in both 1.x and 2.0 so the extra logic is not required :)

File: recipes/openvino/all/conanfile.py
Comment (prince-chrismc) [inline recipes/openvino/all/conanfile.py:65]: This feature is not documented as far as I can tell so it's probably too early to introduce it in CCI.
I would also say these are not very descriptive and do not add much so I think it fine to remove them.

File: recipes/openvino/all/conanfile.py
Comment (prince-chrismc) [inline recipes/openvino/all/conanfile.py:169]: version ranges are still new and there are some extra rules that exist :) https://github.com/conan-io/conan-center-index/blob/46c64a6ce06c8bc97ffdff8a96ef694069103fc4/docs/adding_packages/dependencies.md?plain=1#L176

**Review by CJCombrink — APPROVED** [review]

File: recipes/openvino/all/conandata.yml
Comment (samuel-emrys) [inline recipes/openvino/all/conandata.yml:15]: I could be wrong (maintainer please chime in here!), but my understanding is that dependencies can't be vendored as source code like this for CCI (assuming I understand your intent here correctly). My understanding is that these need to be modelled as requirements, so each would need its own recipe.

Additionally, I'm not sure what the current state of oneDNN is - it looks like it might be open source now? At one point it was closed source or its license was prohibitive, and so it got a special mention in https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#how-to-package-libraries-that-depend-on-proprietary-closed-source-libraries

If it is still closed source or proprietary, I don't think it would be packageable on CCI.

Comment (ilya-lavrenov) [reply recipes/openvino/all/conandata.yml:15]: > My understanding is that these need to be modelled as requirements, so each would need its own recipe.

I agree about oneDNN for GPU and ARM Computer Library, because they are official repositories. But there are not such recipes in Conan for now and it's a simplest way to enable OpenVINO recipe.

In future, once me or someone else will add such recipes, we can re-use it here. But I suppose it's not a blocker to merge current recipe. 

> Additionally, I'm not sure what the current state of oneDNN is - it looks like it might be open source now? At one point it was closed source or its license was prohibitive

It's Apache 2.0, just check "about" section of the repo https://github.com/oneapi-src/oneDNN

Some others dependencies (like MLAS) are pure OpenVINO internal code, which is just located in a submodule for OpenVINO repo for better modularity. But we don't have a goal to distribute it as a separate product / component.

Comment (samuel-emrys) [reply recipes/openvino/all/conandata.yml:15]: > In future, once me or someone else will add such recipes, we can re-use it here. But I suppose it's not a blocker to merge current recipe.

It might be a blocker - I've had PRs blocked on this in the past. A maintainer will have to make a decision on that.

> It's Apache 2.0, just check "about" section of the repo https://github.com/oneapi-src/oneDNN

Yes, I think what's confused me in the past has been that for some of the one* libraries the interfaces were open source but the implementations weren't. I think this applied to oneMKL but it wasn't clear to me how this applied to other libraries in the family.

> Some others dependencies (like MLAS) are pure OpenVINO internal code, which is just located in a submodule for OpenVINO repo for better modularity. But we don't have a goal to distribute it as a separate product / component.

That's good context to have, thanks for clarifying. Does the MLAS dependency you've got share any commonality with https://github.com/microsoft/onnxruntime/tree/main/onnxruntime/core/mlas?

[This statement](https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#handling-internal-dependencies) indicates that it might need a package to avoid ODR violations if symbols haven't been deconflicted with upstream. There's no package in CCI for mlas at the moment so I'm not sure how much of a risk this poses, but there is one for onnxruntime, so maybe there's a chance for conflicts?

Comment (ilya-lavrenov) [reply recipes/openvino/all/conandata.yml:15]: > so maybe there's a chance for conflicts?

Yes, but OpenVINO and ONNX Runtime are both for the same tasks, so customers usually have to select which inference framework to use and go to production only with one of them. So, on practice such conflict is unlikely to happen.

File: recipes/openvino/all/conanfile.py
Comment (ilya-lavrenov) [reply recipes/openvino/all/conanfile.py:169]: is it allowed to use version ranges at least for tools?

File: recipes/openvino/all/conanfile.py
Comment (ilya-lavrenov) [reply recipes/openvino/all/conanfile.py]: fixed

File: recipes/openvino/all/conanfile.py
Comment (ilya-lavrenov) [reply recipes/openvino/all/conanfile.py]: fixed

File: recipes/openvino/all/conanfile.py
Comment (ilya-lavrenov) [reply recipes/openvino/all/conanfile.py]: removed

**Comment by ilya-lavrenov** [comment]
Hi @SpaceIm @RubenRBS @jcar87,
Please, review the recipe. 

It would be great to review and merge the recipe soon, to be aligned with OpenVINO 2023.1.0 official release.

File: recipes/openvino/all/conanfile.py
Comment (RubenRBS) [reply recipes/openvino/all/conanfile.py:194]: )If im reading the upstream cml correctly, that is)

**Review by RubenRBS — COMMENTED** [review]
Thanks a lot! The recipe looks great as far as I can see - I only have minor comments but nothing blocking :)

I do want to bring the conandata.yml discussion up with the team, let's find the best way forward

File: recipes/openvino/all/conandata.yml
Comment (RubenRBS) [reply recipes/openvino/all/conandata.yml:15]: I've slotted this topic for our Monday/Tuesday meeting, thanks for your patience :)

File: recipes/openvino/all/conanfile.py
Comment (ilya-lavrenov) [reply recipes/openvino/all/conanfile.py:194]: `system` means wider than just Conan cache :)

Such options are used to control whether to use submodules inside OpenVINO repo or `find_package` which can find packages in brew, apt, Conan, yum, conda-forge infrastructure.

File: recipes/openvino/all/conandata.yml
Comment (ilya-lavrenov) [reply recipes/openvino/all/conandata.yml:15]: Ok, note, that we are OK to improve the recipe over the time. 
Our vision is to split the work into parts. We want to add OpenVINO - let's add OpenVINO first, its dependencies later, not in opposite way :)
Currently, those dependencies are supposed to be submodules. So, for 2023.1 it's the only way. But once ARM Compute Library / oneDNN recipe is out, we can add such type of source into our cmake infra and consume packages from Conan cache as well.

Comment (memsharded) [reply recipes/openvino/all/conandata.yml:15]: I am perfectly fine with the maintainers doing a release of ``openvino`` based on 3 different sources repositories, with their own commits, etc. That is not vendoring of dependencies, they are releasing their ``openvino`` the way they intend to release it and be used by consumers of it. The fact that there are 3 different repos is an internal implementation detail, and it is not fair or necessary to force them to do separate package releases for every source repo. If they chose to do it in the future, that is good, but shouldn't be mandated from Conan side. 

OneDNN coming from the same org, is a bit in the limit of the vendoring definition, but coming from the same org, I am also fine with initially allowing it, and postponing a bit the extraction as independent package, being this the recommended path.

A different story is the ``arm_compute``  package, that looks like a vendoring of a third party. Even if the result is fully embedded and isolated, at the very least it could be causing duplication of efforts for someone else that might want to use ``arm_compute``. I'd recommend to extract this one in its own package, as we have been doing in the past, unless there are some big challenges or blockers to make ``arm_compute`` its own package, if you need help or can't do it now, we might be able to try to help and contribute it.

(Note: If we have had cases in the past where authors and library maintainers were not allowed to vendor their own organization sources and forced to release independent packages, we should revisit that, learned why it happened, and try to improve and clarify for the future. There are many things in ConanCenter that are not immutable truths, things evolve with the experience, knowledge and changes in the community and ecosystem)

Comment (ilya-lavrenov) [reply recipes/openvino/all/conandata.yml:15]: > I'd recommend to extract this one in its own package

I agree with such position, but am I right that it's not a blocker to merge? 
It can be done later as improvement. 

(BTW, I have already made 19 PRs to CCI which got merged, so I want to finally merge our recipe and continue improvements after merge)

**Review by RubenRBS — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
Hi @RubenRBS 
I really appreciate your review. I definitely migrate to Compute Library once its recipe will be a part of CCI.
Could you please let me know whether we need review from someone else? If so, could you please invite those people?

**Comment by RubenRBS** [comment]
Only 1 team member review + 1 community review is needed for this to get merged, but CCI is currently under heavy load which might delay the automatic merge a bit (No new merges have happened in the last 20h or so) we expect things to move again early morning tomorrow

File: recipes/openvino/all/conanfile.py
Comment (uilianries) [inline recipes/openvino/all/conanfile.py:257]: When say `clang` does it include Clang from MacOS too?

Comment (ilya-lavrenov) [reply recipes/openvino/all/conanfile.py:257]: OpenVINO GPU plugin is not available on macOS

**Review by uilianries — APPROVED** [review]
The ConanCenterIndex when running Conan 2.x it uses the combination `*/*:shared=true` or `*/*:shaed=False`, which result in invalid configuration for any build, because both Protobuf and OneTBB require a mixed configuration.

Plus, we don't have cross-building profile to check if it works with ARM. So, to prove that's working on Linux-ARM, I cross-built to armv8. It passed without errors. Full log:

[openvino_build.log](https://github.com/conan-io/conan-center-index/files/12729445/openvino_build.log)

About Compute Library, there is a separate PR, but will take more time to pass, and will need changes in this recipe. So, I consider this PR as good enough to be approved now and be improved as soon as we have Compute Library.

File: recipes/openvino/all/conanfile.py
Comment (SpaceIm) [inline recipes/openvino/all/conanfile.py:139]: @RubenRBS @uilianries I though that version range was only allowed for cmake, openssl, libcurl and zlib.

File: recipes/openvino/all/conanfile.py
Comment (SpaceIm) [inline recipes/openvino/all/conanfile.py:146]: version ranges must have an upper bound according to conancenter documentation

File: recipes/openvino/all/conanfile.py
Comment (uilianries) [reply recipes/openvino/all/conanfile.py:139]: Correct, it was accepted by mistake!

Comment (RubenRBS) [reply recipes/openvino/all/conanfile.py:139]: You're totally right. We're thinking about allowing it for every tool_require in the future, but we're not there yet, thanks for noticing - This PR got sidetracked and we merged before we should have, and things like this slipped by - Pinging @ilya-lavrenov for when the update to use arm_compute is done, could you also please fix this? Thanks!

Comment (ilya-lavrenov) [reply recipes/openvino/all/conanfile.py:139]: I supposed that for tools we can use ranges (and about it here https://github.com/conan-io/conan-center-index/pull/19681#discussion_r1326432158)

Ok, I will pin tools version as well in my nearest update (I plan to backport some patches from upstream to make recipe more future proof).

File: recipes/openvino/all/conanfile.py
Comment (ilya-lavrenov) [reply recipes/openvino/all/conanfile.py:146]: fixed in https://github.com/conan-io/conan-center-index/pull/20245

File: recipes/openvino/all/conanfile.py
Comment (ilya-lavrenov) [reply recipes/openvino/all/conanfile.py:139]: fixed in https://github.com/conan-io/conan-center-index/pull/20245

**Pull request merged**
